### ISSUE TITLE
TypeScript Error: 'MentionsInput' and 'Mention' Components Not Compatible as JSX Elements

### ISSUE THREAD

**Issue by haslam790** [signavio/react-mentions#767]
**Steps to reproduce:**

1. Set up a TypeScript React project with the MentionsInput and Mention components.
2. Attempt to use the MentionsInput and Mention components within a .tsx file.
3. Compile the project using tsc or run the app.
4. Observe the error in the terminal/logs.

**Expected behaviour:**
MentionsInput and Mention should render without any TypeScript errors.
**Observed behaviour:**
TypeScript throws the following errors:

Error TS2786: 'MentionsInput' cannot be used as a JSX component.

Explanation: Its type 'MentionsInputClass' is not a valid JSX element type.
Details: Type 'ComponentClass<MentionsInputProps, any>' is not assignable to type 'new (props: any, deprecatedLegacyContext?: any) => Component<any, any, any>'. Property 'refs' is missing in type 'Component<MentionsInputProps, any, any>' but required in type 'Component<any, any, any>'.
Error TS2786: 'Mention' cannot be used as a JSX component.

Explanation: Its type `'FC<MentionProps>' is not a valid JSX element type.
Details: Type 'FunctionComponent<MentionProps>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
**Workaround:**
No workaround identified as of now.

**Comment by prithwish-witbybit** [comment]
I'm facing the same issue. Is there any update?

**Comment by ebarthur** [comment]
Did you find a solution? Or at least an alternative? @prithwish-witbybit

**Comment by prithwish-witbybit** [comment]
Yes, I'm creating a new variable and changing the component type to `any`. Then using that component in JSX
![Image](https://github.com/user-attachments/assets/543d9e49-93f5-4dd2-897e-bb88ca30fa24)

![Image](https://github.com/user-attachments/assets/76f88faa-e49c-4293-81e6-6d38feec83d0)

**Comment by hbmartin** [comment]
This package looks abandon but I am working to resurrect with a modern rewrite - fully rewritten in TS.
You can try it today (but it's not 💯 and I'm still evolving it):
https://github.com/hbmartin/react-mentions-ts
